apps/comps/rpc/router/user/link-wallet.ts

```diff
@@ -0,0 +1,73 @@
+import { ORPCError } from "@orpc/server";
+import { z } from "zod";
+
+import { verifyPrivyUserHasLinkedWallet } from "@recallnet/services/lib";
+
+import { base } from "@/rpc/context/base";
+import { authMiddleware } from "@/rpc/middleware/auth";
+import { serializeUser } from "@/rpc/router/utils/serialize-user";
+
+export const linkWallet = base
+  .use(authMiddleware)
+  .input(z.object({ walletAddress: z.string() }))
+  .handler(async ({ input, context, errors }) => {
+    try {
+      // Get Privy identity token from cookies
+      const idToken = context.cookies.get("privy-id-token")?.value;
+      if (!idToken) {
+        throw errors.UNAUTHORIZED();
+      }
+
+      // Verify the wallet is linked in Privy
+      const isLinked = await verifyPrivyUserHasLinkedWallet(
+        idToken,
+        context.privyClient,
+        input.walletAddress,
+      );
+
+      if (!isLinked) {
+        throw errors.BAD_REQUEST({ message: "Wallet not linked to user" });
+      }
+
+      // Update user with the new wallet address and verification timestamp
+      const now = new Date();
+      const updatedUser = await context.userService.updateUser({
+        id: context.user.id,
+        walletAddress: input.walletAddress,
+        walletLastVerifiedAt: now,
+        updatedAt: now,
+      });
+
+      // Grant initial boost if in stake mode
+      const isStakeMode = !process.env.NEXT_PUBLIC_NON_STAKE_BOOST_AMOUNT;
+      if (isStakeMode) {
+        await context.boostAwardService.initForStake(updatedUser.walletAddress);
+      }
```

Comment:
Leave this for now, but I really like the idea of just keeping the "start boosting" button on competition pages, then there is no need for this side effect.

Reply:
Let's talk about that as a team and decide.